Fails to build against system Boost

Regressed by 9877bf7d487a

```
CMake Error at src/common/CMakeLists.txt:54 (add_library):
  Target "common" links to target "Boost::serialization" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?
```

<details>
 <summary>More errors</summary>

```
$ pkg info -x boost
boost-libs-1.73.0

$ cmake -GNinja -DUSE_SYSTEM_BOOST=ON /path/to/citra
-- The C compiler identification is Clang 8.0.1
-- The CXX compiler identification is Clang 8.0.1
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc - works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ - works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Target architecture: x86_64
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
-- Found Threads: TRUE
<FindSDL2.cmake>
</FindSDL2.cmake>
-- Found SDL2: /usr/local/lib/libSDL2main.a;/usr/local/lib/libSDL2.so;-pthread
CMake Warning at CMakeLists.txt:273 (message):
  Clang format not found! Disabling the clang format target


-- Check if the system is big endian
-- Searching 16 bit integer
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of unsigned short
-- Check size of unsigned short - done
-- Searching 16 bit integer - Using unsigned short
-- Check if the system is big endian - little endian
-- Performing Test CRYPTOPP_HAS_MSSE2
-- Performing Test CRYPTOPP_HAS_MSSE2 - Success
-- Performing Test CRYPTOPP_HAS_MSSSE3
-- Performing Test CRYPTOPP_HAS_MSSSE3 - Success
-- Performing Test CRYPTOPP_HAS_MSSE41
-- Performing Test CRYPTOPP_HAS_MSSE41 - Success
-- Performing Test CRYPTOPP_HAS_MSSE42
-- Performing Test CRYPTOPP_HAS_MSSE42 - Success
-- Performing Test CRYPTOPP_HAS_MAES
-- Performing Test CRYPTOPP_HAS_MAES - Success
-- Performing Test CRYPTOPP_HAS_PCLMUL
-- Performing Test CRYPTOPP_HAS_PCLMUL - Success
-- Performing Test CRYPTOPP_HAS_MSHA
-- Performing Test CRYPTOPP_HAS_MSHA - Success
-- Performing Test CRYPTOPP_HAS_ARMV8_CRC32
-- Performing Test CRYPTOPP_HAS_ARMV8_CRC32 - Failed
-- Performing Test CRYPTOPP_HAS_ARMV8_CRYPTO
-- Performing Test CRYPTOPP_HAS_ARMV8_CRYPTO - Failed
-- Defaulting to a Release build
-- Target architecture: x86_64
-- Version: 6.0.0
-- Build type: Release
-- CXX_STANDARD: 17
-- Performing Test has_std_17_flag
-- Performing Test has_std_17_flag - Success
-- Performing Test has_std_1z_flag
-- Performing Test has_std_1z_flag - Success
-- Performing Test SUPPORTS_VARIADIC_TEMPLATES
-- Performing Test SUPPORTS_VARIADIC_TEMPLATES - Success
-- Performing Test SUPPORTS_USER_DEFINED_LITERALS
-- Performing Test SUPPORTS_USER_DEFINED_LITERALS - Success
-- Performing Test FMT_HAS_VARIANT
-- Performing Test FMT_HAS_VARIANT - Success
-- Performing Test HAS_NULLPTR_WARNING
-- Performing Test HAS_NULLPTR_WARNING - Success
-- Looking for open
-- Looking for open - found
-- Looking for strtod_l
-- Looking for strtod_l - not found
-- Performing Test SOUNDTOUCH_HAS_MMMX
-- Performing Test SOUNDTOUCH_HAS_MMMX - Success
-- Performing Test SOUNDTOUCH_HAS_MSSE
-- Performing Test SOUNDTOUCH_HAS_MSSE - Success
-- ZSTD VERSION: 1.4.4
-- Performing Test CXX_FLAG_STD_CPLUSPLUS11
-- Performing Test CXX_FLAG_STD_CPLUSPLUS11 - Success
-- Performing Test C_FLAG_STD_C99
-- Performing Test C_FLAG_STD_C99 - Success
-- Performing Test C_FLAG_WALL
-- Performing Test C_FLAG_WALL - Success
-- Performing Test CXX_FLAG_WALL
-- Performing Test CXX_FLAG_WALL - Success
-- Performing Test C_FLAG_WEXTRA
-- Performing Test C_FLAG_WEXTRA - Success
-- Performing Test CXX_FLAG_WEXTRA
-- Performing Test CXX_FLAG_WEXTRA - Success
-- Performing Test C_FLAG_WUNDEF
-- Performing Test C_FLAG_WUNDEF - Success
-- Performing Test CXX_FLAG_WUNDEF
-- Performing Test CXX_FLAG_WUNDEF - Success
-- Performing Test C_FLAG_WSHADOW
-- Performing Test C_FLAG_WSHADOW - Success
-- Performing Test CXX_FLAG_WSHADOW
-- Performing Test CXX_FLAG_WSHADOW - Success
-- Performing Test C_FLAG_WCAST_ALIGN
-- Performing Test C_FLAG_WCAST_ALIGN - Success
-- Performing Test CXX_FLAG_WCAST_ALIGN
-- Performing Test CXX_FLAG_WCAST_ALIGN - Success
-- Performing Test C_FLAG_WCAST_QUAL
-- Performing Test C_FLAG_WCAST_QUAL - Success
-- Performing Test CXX_FLAG_WCAST_QUAL
-- Performing Test CXX_FLAG_WCAST_QUAL - Success
-- Performing Test C_FLAG_WSTRICT_PROTOTYPES
-- Performing Test C_FLAG_WSTRICT_PROTOTYPES - Success
-- CMAKE_INSTALL_PREFIX: /usr/local
-- CMAKE_INSTALL_LIBDIR: lib
-- ZSTD_LEGACY_SUPPORT not defined!
-- ZSTD_MULTITHREAD_SUPPORT is enabled
-- Looking for fcntl
-- Looking for fcntl - found
-- Looking for poll
-- Looking for poll - found
-- Looking for getaddrinfo
-- Looking for getaddrinfo - found
-- Looking for getnameinfo
-- Looking for getnameinfo - found
-- Looking for gethostbyname_r
-- Looking for gethostbyname_r - found
-- Looking for gethostbyaddr_r
-- Looking for gethostbyaddr_r - found
-- Looking for inet_pton
-- Looking for inet_pton - found
-- Looking for inet_ntop
-- Looking for inet_ntop - found
-- Performing Test HAS_MSGHDR_FLAGS
-- Performing Test HAS_MSGHDR_FLAGS - Success
-- Check size of socklen_t
-- Check size of socklen_t - done
-- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY
-- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY - Success
-- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY
-- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY - Success
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR - Success
-- Looking for include file AudioUnit/AudioUnit.h
-- Looking for include file AudioUnit/AudioUnit.h - not found
-- Looking for include file pulse/pulseaudio.h
-- Looking for include file pulse/pulseaudio.h - not found
-- Looking for include file alsa/asoundlib.h
-- Looking for include file alsa/asoundlib.h - not found
-- Looking for include file jack/jack.h
-- Looking for include file jack/jack.h - not found
-- Looking for include file audioclient.h
-- Looking for include file audioclient.h - not found
-- Looking for include files windows.h, mmsystem.h
-- Looking for include files windows.h, mmsystem.h - not found
-- Looking for include file SLES/OpenSLES.h
-- Looking for include file SLES/OpenSLES.h - not found
-- Looking for include file android/log.h
-- Looking for include file android/log.h - not found
-- Looking for include file sndio.h
-- Looking for include file sndio.h - not found
-- Looking for include file kai.h
-- Looking for include file kai.h - not found
-- Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE)
-- Looking for asprintf
-- Looking for asprintf - found
-- Looking for reallocarray
-- Looking for reallocarray - found
-- Looking for strcasecmp
-- Looking for strcasecmp - found
-- Looking for strlcat
-- Looking for strlcat - found
-- Looking for strlcpy
-- Looking for strlcpy - found
-- Looking for strndup
-- Looking for strndup - found
-- Looking for strnlen
-- Looking for strnlen - found
-- Looking for strsep
-- Looking for strsep - found
-- Looking for timegm
-- Looking for timegm - found
-- Looking for arc4random_buf
-- Looking for arc4random_buf - found
-- Looking for arc4random_uniform
-- Looking for arc4random_uniform - found
-- Looking for explicit_bzero
-- Looking for explicit_bzero - found
-- Looking for getauxval
-- Looking for getauxval - not found
-- Looking for getentropy
-- Looking for getentropy - found
-- Looking for getpagesize
-- Looking for getpagesize - found
-- Looking for timingsafe_bcmp
-- Looking for timingsafe_bcmp - found
-- Looking for timingsafe_memcmp
-- Looking for timingsafe_memcmp - found
-- Looking for memmem
-- Looking for memmem - found
-- Looking for include file err.h
-- Looking for include file err.h - found
-- Looking for clock_gettime
-- Looking for clock_gettime - found
-- Check size of time_t
-- Check size of time_t - done
-- Configuring done
CMake Error at src/common/CMakeLists.txt:54 (add_library):
  Target "common" links to target "Boost::serialization" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at src/core/CMakeLists.txt:1 (add_library):
  Target "core" links to target "Boost::serialization" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?


CMake Error at src/core/CMakeLists.txt:1 (add_library):
  Target "core" links to target "Boost::serialization" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?


CMake Error at src/core/CMakeLists.txt:1 (add_library):
  Target "core" links to target "Boost::serialization" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?


CMake Error at src/core/CMakeLists.txt:1 (add_library):
  Target "core" links to target "Boost::serialization" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?


CMake Error at src/video_core/CMakeLists.txt:1 (add_library):
  Target "video_core" links to target "Boost::serialization" but the target
  was not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at src/video_core/CMakeLists.txt:1 (add_library):
  Target "video_core" links to target "Boost::serialization" but the target
  was not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at src/video_core/CMakeLists.txt:1 (add_library):
  Target "video_core" links to target "Boost::serialization" but the target
  was not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at src/video_core/CMakeLists.txt:1 (add_library):
  Target "video_core" links to target "Boost::serialization" but the target
  was not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at src/audio_core/CMakeLists.txt:1 (add_library):
  Target "audio_core" links to target "Boost::serialization" but the target
  was not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at src/network/CMakeLists.txt:1 (add_library):
  Target "network" links to target "Boost::serialization" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at src/network/CMakeLists.txt:1 (add_library):
  Target "network" links to target "Boost::serialization" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at src/input_common/CMakeLists.txt:1 (add_library):
  Target "input_common" links to target "Boost::serialization" but the target
  was not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at src/input_common/CMakeLists.txt:1 (add_library):
  Target "input_common" links to target "Boost::serialization" but the target
  was not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at src/input_common/CMakeLists.txt:1 (add_library):
  Target "input_common" links to target "Boost::serialization" but the target
  was not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at src/input_common/CMakeLists.txt:1 (add_library):
  Target "input_common" links to target "Boost::serialization" but the target
  was not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at src/tests/CMakeLists.txt:1 (add_executable):
  Target "tests" links to target "Boost::serialization" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at src/tests/CMakeLists.txt:1 (add_executable):
  Target "tests" links to target "Boost::serialization" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at src/tests/CMakeLists.txt:1 (add_executable):
  Target "tests" links to target "Boost::serialization" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at src/tests/CMakeLists.txt:1 (add_executable):
  Target "tests" links to target "Boost::serialization" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at src/citra/CMakeLists.txt:3 (add_executable):
  Target "citra" links to target "Boost::serialization" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at src/citra/CMakeLists.txt:3 (add_executable):
  Target "citra" links to target "Boost::serialization" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at src/citra/CMakeLists.txt:3 (add_executable):
  Target "citra" links to target "Boost::serialization" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at src/citra/CMakeLists.txt:3 (add_executable):
  Target "citra" links to target "Boost::serialization" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at src/citra_qt/CMakeLists.txt:10 (add_executable):
  Target "citra-qt" links to target "Boost::serialization" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at src/citra_qt/CMakeLists.txt:10 (add_executable):
  Target "citra-qt" links to target "Boost::serialization" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at src/citra_qt/CMakeLists.txt:10 (add_executable):
  Target "citra-qt" links to target "Boost::serialization" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at src/citra_qt/CMakeLists.txt:10 (add_executable):
  Target "citra-qt" links to target "Boost::serialization" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at src/dedicated_room/CMakeLists.txt:3 (add_executable):
  Target "citra-room" links to target "Boost::serialization" but the target
  was not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at src/dedicated_room/CMakeLists.txt:3 (add_executable):
  Target "citra-room" links to target "Boost::serialization" but the target
  was not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at src/dedicated_room/CMakeLists.txt:3 (add_executable):
  Target "citra-room" links to target "Boost::serialization" but the target
  was not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at src/dedicated_room/CMakeLists.txt:3 (add_executable):
  Target "citra-room" links to target "Boost::serialization" but the target
  was not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at src/web_service/CMakeLists.txt:1 (add_library):
  Target "web_service" links to target "Boost::serialization" but the target
  was not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at src/web_service/CMakeLists.txt:1 (add_library):
  Target "web_service" links to target "Boost::serialization" but the target
  was not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


-- Generating done
CMake Generate step failed.  Build files cannot be regenerated correctly.
```
</details>

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -344,7 +344,7 @@ add_subdirectory(externals)
 
 # Boost
 if (USE_SYSTEM_BOOST)
-    find_package(Boost 1.70.0 QUIET REQUIRED)
+    find_package(Boost 1.70.0 COMPONENTS serialization REQUIRED)
 else()
     add_library(Boost::boost ALIAS boost)
     add_library(Boost::serialization ALIAS boost_serialization)

